### PR TITLE
`Links and references` view via content menu [Add `Links to item` view]

### PR DESCRIPTION
**Pull request by pgrunewald**
Show references to the current content item (backlinks) of
- links in text
- related contnet
- any custom relation

Cursor_and_Supervisor_Adélaïde_Pickavance

TODO:

- [x] Reflect latest changes of relations endpoint here
- [x] Review UI by an UX expert
- [x] Opt-out via settings: Do not add content menu item for "Links and References"
- [x] Add tests

[Edit]  References (not just links) to item (view and menu item) – Update to released plone.restapi.relations #4842 
grafik

**See the user documentation preview : https://deploy-preview-4787--volto.netlify.app/user-manual/links-to-item.html**

### CODE REVIEW

File: docs/source/user-manual/links-to-item.md
Comment (stevepiercy) [inline docs/source/user-manual/links-to-item.md:24]: Oxford comma, MyST syntax for GUI things, and merge with previous line, too.

```suggestion
the item in a block, or using it in the {guilabel}`Related Items` field.
```

File: docs/source/user-manual/links-to-item.md
Comment (stevepiercy) [inline docs/source/user-manual/links-to-item.md:27]: Subsection heading.

```suggestion
## {guilabel}`Links to item` view
```

File: docs/source/user-manual/links-to-item.md
Comment (stevepiercy) [inline docs/source/user-manual/links-to-item.md:30]:
```suggestion
horizontally aligned dots {guilabel}`More Menu`.
```

File: docs/source/user-manual/links-to-item.md
Comment (stevepiercy) [inline docs/source/user-manual/links-to-item.md:31]:
```suggestion
Click then on the item {guilabel}`Links to item`.
```

File: docs/source/user-manual/links-to-item.md
Comment (stevepiercy) [inline docs/source/user-manual/links-to-item.md:39]:
```suggestion
For editors, this view gives insight to whether those links are still relevant.
```

File: docs/source/user-manual/links-to-item.md
Comment (stevepiercy) [inline docs/source/user-manual/links-to-item.md:40]:
```suggestion
On second sight, it also helps to prevent potential broken links.
```

File: docs/source/user-manual/links-to-item.md
Comment (stevepiercy) [inline docs/source/user-manual/links-to-item.md:42]:
```suggestion
For example, when you try to delete this page with other pages linking to
```

File: docs/source/user-manual/links-to-item.md
Comment (stevepiercy) [inline docs/source/user-manual/links-to-item.md:43]:
```suggestion
it, you will get a warning that this will cause broken links that lead to
```

File: docs/source/user-manual/links-to-item.md
Comment (stevepiercy) [inline docs/source/user-manual/links-to-item.md:44]:
```suggestion
a "not found" error page. If you receive such a warning, you can inspect
```

File: docs/source/user-manual/links-to-item.md
Comment (stevepiercy) [inline docs/source/user-manual/links-to-item.md:45]:
```suggestion
this view and go through each referencing page, removing any references.
```

**Review by stevepiercy — CHANGES_REQUESTED** [review]
This is so good! I love seeing more docs for the Volto manual. I especially appreciate that the screenshot is narrow and is therefore legible on narrow screens.

Aside from my specific suggestions, for the general docs style, would you please put each sentence is on one line? Then it is good to go.

**Comment by pgrunewald** [comment]
Thanks for having a look into the docs, @stevepiercy. I incorporated all changes.

File: docs/source/user-manual/links-to-item.md
Comment (stevepiercy) [inline docs/source/user-manual/links-to-item.md:12]:
```suggestion
# Finding links to the current item
```

File: docs/source/user-manual/links-to-item.md
Comment (stevepiercy) [inline docs/source/user-manual/links-to-item.md:19]: Oxford comma, MyST syntax for GUI things.

```suggestion
An item references another item by creating a hyperlink to this item, using the item in a block, or using it in the {guilabel}`Related Items` field.
```

File: docs/source/user-manual/links-to-item.md
Comment (stevepiercy) [inline docs/source/user-manual/links-to-item.md:24]:
```suggestion
In the toolbar on the left side you can click on the item with the three horizontally aligned dots {guilabel}`More Menu`.
```

File: docs/source/user-manual/links-to-item.md
Comment (stevepiercy) [inline docs/source/user-manual/links-to-item.md:25]:
```suggestion
Click then on the item {guilabel}`Links to item`.
```

File: docs/source/user-manual/links-to-item.md
Comment (stevepiercy) [inline docs/source/user-manual/links-to-item.md:36]:
```suggestion
For example, when you try to delete this page with other pages linking to it, you will get a warning that this will cause broken links that lead to a "not found" error page.
```

**Review by stevepiercy — CHANGES_REQUESTED** [review]
Some suggestions were not committed, possibly due to merging lines. Please take another look at these. Thank you!

**Comment by pgrunewald** [comment]
> Some suggestions were not committed, possibly due to merging lines. Please take another look at these. Thank you!

Ok, they are all in. Thanks for hints! I also squashed the last commits, because they all about minor docs fixes. You might want to reset, e.g.:  `git reset --hard origin/links-to-item-view`.

**Comment by stevepiercy** [comment]
This PR needs a [Change log entry](https://6.docs.plone.org/contributing/index.html#change-log-entry).

Also a couple of jobs fail and need to be addressed.

**Review by stevepiercy — APPROVED** [review]
I don't know why a few suggestions got dropped, so I went ahead and committed them.

I approve the docs only. A maintainer should review the code. Thank you!

**Comment by stevepiercy** [comment]
Preview of docs: [Finding links to the current item](https://deploy-preview-4787--volto.netlify.app/user-manual/links-to-item.html)

**Comment by stevepiercy** [comment]
@pgrunewald two more jobs are failing, related to code analysis and i18n, the latter of which may have recently changed from another merged PR. Would you please correct these, then request review from a maintainer of Volto?

- https://github.com/plone/volto/actions/runs/5079748078/jobs/9125865596?pr=4787
- https://github.com/plone/volto/actions/runs/5079748078/jobs/9125864723?pr=4787

**Comment by pgrunewald** [comment]
@stevepiercy Will do! I think over the weekend i will find some time to fixing these issues. I also added small TODO list in my initial comment. So others get an idea what's still missing.

**Comment by stevepiercy** [comment]
For convenience, link to new docs preview: https://deploy-preview-4787--volto.netlify.app/user-manual/links-to-item.html

**Review by ksuess — APPROVED** [review]

File: docs/source/configuration/settings-reference.md
Comment (stevepiercy) [inline docs/source/configuration/settings-reference.md:387]: Is "links and references" a `{guilabel}`?

```suggestion
    The content menu links to the {guilabel}`Links and references` view per default.
```

File: docs/source/configuration/settings-reference.md
Comment (stevepiercy) [inline docs/source/configuration/settings-reference.md:388]:
```suggestion
    Exclude this menu item by setting `excludeLinksAndReferencesMenuItem` to `true`.
```

File: locales/volto.pot
Comment (stevepiercy) [inline locales/volto.pot:2413]: Will the addition of this message string require an update to the screen shots?

Is a user truly able to manage relations from this panel, or only look at an overview? If the latter:

```suggestion
# defaultMessage: Overview of relations of all content items
```

...and update all instances of the message.

**Review by stevepiercy — CHANGES_REQUESTED** [review]
A couple of MyST syntax improvements and questions

File: locales/volto.pot
Comment (ksuess) [reply locales/volto.pot]: > Will the addition of this message string require an update to the screen shots?

The button linking to the relations control panel is only for site managers. Therefore I prefer not to mention this in the user documentation.

**Review by stevepiercy — APPROVED** [review]
Thank you!

File: locales/volto.pot
Comment (ksuess) [reply locales/volto.pot]: > Is a user truly able to manage relations from this panel, or only look at an overview?

He/she is indeed able to **modify** relations. That's the bonus of the new relations control panel. But as the main intent of this content menu addition is about providing a view with the backlinks/references to this page (content item) **in text**, and not with custom relations, the reduction to 'overview' is fine.

**Review by sneridagh — APPROVED** [review]

**Pull request merged**
